vaadin-grid-flow/src/main/resources/META-INF/resources/frontend/gridConnector.js

```diff
@@ -15,7 +15,15 @@ import { ItemCache } from '@vaadin/vaadin-grid/src/vaadin-grid-data-provider-mix
         return;
       }
 
+      // Storing original implementation of the method to be used for client
+      // side only grids
+      ItemCache.prototype.ensureSubCacheForScaledIndexOriginal = ItemCache.prototype.ensureSubCacheForScaledIndex;
       ItemCache.prototype.ensureSubCacheForScaledIndex = tryCatchWrapper(function(scaledIndex) {
```

Comment:
Also noticed that these changes to `ItemCache` prototype are done on every grid added to the page. I thought of adding a check for it to be initialized only once (but keeping it along aside the first grid initialization). I wasn't sure if that would have any impact (good or bad) to the component, so I didn't add, but if we think that this would be beneficial, we could bring this change in another PR.

Reply:
Sounds like a good idea. Maybe it would work if the prototype patching was moved outside of the `initLazy` function.

Reply:
The idea of keeping it inside of `initLazy` was to delay the patching to the moment where it's needed (once a server-side grid is first attached). But I believe that would be the majority of cases, so it might not be necessary to postpone the prototype patching.